src/js/tabs/onboard_logging.js

```
@@ -489,49 +489,59 @@ onboard_logging.initialize = function (callback) {
-                                // A zero-byte block indicates end-of-file, so we're done
-                                mark_saving_dialog_done(startTime, nextAddress, totalBytesCompressed);
-                                FileSystem.closeFile(openedFile);
-                            }
-                        } else {
-                            // There was an error with the received block (address didn't match the one we asked for), retry
-                            mspHelper.dataflashRead(nextAddress, self.blockSize, onChunkRead);
-                        }
-                    }
+const activeRetries = new Set(); // tracks blocks currently retrying
+
+async function onChunkRead(blockAddr, blockData) {
```

ESLint

[error] Expected indentation of 20 spaces but found 0.

(indent)